tests/ProjectTypeResolverTest.php

```diff
@@ -12,25 +12,20 @@
 use Composer\Composer;
 use Composer\Config;
 use Composer\Package\RootPackageInterface;
+use PHPUnit\Framework\Attributes\CoversMethod;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\MockObject\Exception;
 use PHPUnit\Framework\TestCase;
 use Youwe\TestingSuite\Composer\ProjectTypeResolver;
 
-/**
- * @coversDefaultClass \Youwe\TestingSuite\Composer\ProjectTypeResolver
- */
+#[CoversMethod(ProjectTypeResolver::class, '__construct')]
+#[CoversMethod(ProjectTypeResolver::class, 'resolve')]
 class ProjectTypeResolverTest extends TestCase
 {
     /**
-     * @param string $packageType
-     * @param string $expected
-     *
-     * @return void
-     *
-     * @dataProvider dataProvider
-     *
-     * @covers ::__construct
-     * @covers ::resolve
+     * @throws Exception
      */
+    #[DataProvider('dataProvider')]
```

Comment:
Tip: if the data provider is short / not to complex and not reused then https://docs.phpunit.de/en/12.0/attributes.html#testwith is also a nice option